### PR TITLE
added login and register page

### PR DESCRIPTION
**Pull request by kentblock**

### CODE REVIEW

File: fe/src/components/Login.vue [hunk 1]
```diff
@@ -1,19 +1,20 @@
 <template>
   <div class="login">
     <el-row>
-      <el-col offset=9 span=6>
+      <el-col :offset="9" :span="6">
         <h1>Very Cool Stock Application</h1>
-        <el-form status-icon :model="userCredentials">
+        <el-form status-icon :model="userCredentials" ref='LoginForm'>
```
Comment (deathbask3t) [inline fe/src/components/Login.vue]: make LoginForm camelcase

File: fe/src/components/Login.vue [hunk 2]
```diff
@@ -34,12 +35,17 @@ export default {
...
   methods: {
     submitForm() {
-      // not sure what to do here 
-      var res = httpService.post('api/user/create/', {email:'test@email.ca', password:'password'});
-      console.log(res)
-      console.log(res.data)
 
-    }  	
+      this.$refs['LoginForm'].validate((valid) => {
+        if(valid){
```
Comment (deathbask3t) [inline fe/src/components/Login.vue]: spaces after if/else and before braces

File: fe/src/components/Login.vue [hunk 3]
```diff
@@ -34,12 +35,17 @@ export default {
...
+          alert('Logging in')
+        }else{
+          alert('not logging in')
```
Comment (deathbask3t) [inline fe/src/components/Login.vue]: can this be more descriptive? is there an error or another function parameter we can provide to give the user more information?

File: fe/src/components/Login.vue [hunk 4]
```diff
@@ -34,12 +35,17 @@ export default {
...
     submitForm() {
-      // not sure what to do here 
-      var res = httpService.post('api/user/create/', {email:'test@email.ca', password:'password'});
-      console.log(res)
-      console.log(res.data)
 
-    }  	
+      this.$refs['LoginForm'].validate((valid) => {
+        if(valid){
+          alert('Logging in')
```
Comment (deathbask3t) [inline fe/src/components/Login.vue]: If a user successfully logs in its a bit janky that they get a box they have to click through. 

consider https://element.eleme.io/#/en-US/component/notification

File: fe/src/components/Register.vue [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+</template>
+
+<script>
+import httpService from '../services/http.js';
+export default {
+  name: "Register",
+  data () {
+    return {
+      userCredentials: {
+        name:'',
```
Comment (deathbask3t) [inline fe/src/components/Register.vue]: spaces after colon

File: fe/src/components/Register.vue [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        email:'',
+        password:''
+      },
+    }
+  },
+  methods: {
+    submitForm() {
+
```
Comment (deathbask3t) [inline fe/src/components/Register.vue]: delete this space

File: fe/src/components/Register.vue [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+      this.$refs['RegisterForm'].validate((valid) => {
```
Comment (deathbask3t) [inline fe/src/components/Register.vue]: this could be a constant

File: fe/src/components/Login.vue [hunk 8]
```diff
@@ -34,12 +35,17 @@ export default {
   },
   methods: {
     submitForm() {
-      // not sure what to do here 
-      var res = httpService.post('api/user/create/', {email:'test@email.ca', password:'password'});
-      console.log(res)
-      console.log(res.data)
 
-    }  	
+      this.$refs['LoginForm'].validate((valid) => {
```
Comment (deathbask3t) [inline fe/src/components/Login.vue]: this could be a constant

File: fe/src/components/Login.vue [hunk 9]
```diff
@@ -34,12 +35,17 @@ export default {
...
+        if(valid){
+          alert('Logging in')
+        }else{
+          alert('not logging in')
+        }
+      });
+      var res = httpService.post('api/user/create/', this.userCredentials)
```
Comment (deathbask3t) [inline fe/src/components/Login.vue]: one thing i try to do is create a base http service that isn't used in a vue component, but by other services. so the `user.js` service does this bit (with the path as a constant in that service) and then in the code you call `userService.login(this.userCredentials)`

do that

File: fe/src/components/Register.vue [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+        if(valid){
```
Comment (deathbask3t) [inline fe/src/components/Register.vue]: spaces needed

File: fe/src/components/Register.vue [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+          //redirect to log in page
```
Comment (deathbask3t) [inline fe/src/components/Register.vue]: spaces after comments

File: fe/src/components/Register.vue [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+          //make sure account was created succesfully
+          httpService.post('api/user/create/', this.userCredentials).then((response) => {
+            if(response.status == 201){
```
Comment (deathbask3t) [inline fe/src/components/Register.vue]: same thing as above

File: fe/src/components/Register.vue [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+              console.log("Created succesfully")
```
Comment (deathbask3t) [inline fe/src/components/Register.vue]: console.log is developer facing. do this with that notification class i posted above

Comment (deathbask3t) [reply fe/src/components/Register.vue]: Also make this message more descriptive

File: fe/src/components/Register.vue [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+              this.$router.push("Login")
+            }else{
+              console.log("Not created")
+            }
+          }
+          );
+          alert('registering')
```
Comment (deathbask3t) [inline fe/src/components/Register.vue]: notification

File: fe/src/components/Register.vue [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+    submitForm() {
+
+      this.$refs['RegisterForm'].validate((valid) => {
+        if(valid){
+          //redirect to log in page
+          //make sure account was created succesfully
+          httpService.post('api/user/create/', this.userCredentials).then((response) => {
+            if(response.status == 201){
+              console.log("Created succesfully")
+              this.$router.push("Login")
```
Comment (deathbask3t) [inline fe/src/components/Register.vue]: so a user could just type the route into their url bar and go to the stock page. you should add these

https://router.vuejs.org/guide/advanced/navigation-guards.html

File: fe/src/components/Register.vue [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+      
+    }
+  }
+}
+
+</script>
+
+
+<style>
+.register{
```
Comment (deathbask3t) [inline fe/src/components/Register.vue]: fookin spaces man

File: fe/src/main.js [hunk 17]
```diff
@@ -1,11 +1,30 @@
...
 
 Vue.config.productionTip = false
 
+Vue.use(Router)
 Vue.use(ElementUI)
+
+const router = new Router({
+
+  routes: [
+
```
Comment (deathbask3t) [inline fe/src/main.js]: spaces

File: fe/src/main.js [hunk 18]
```diff
@@ -1,11 +1,30 @@
...
 Vue.use(ElementUI)
+
+const router = new Router({
+
+  routes: [
+    
+    {path: '/', component: Home},
+    {path: '/login', name:'Login', component:Login},
+    {path: '/register', component:Register},
+
```
Comment (deathbask3t) [inline fe/src/main.js]: why all these newlines, get rid of them

File: fe/src/services/http.js [hunk 19]
```diff
@@ -22,7 +22,7 @@ var json = (obj) => obj ? JSON.stringify(obj) : ''
 var obj = (json) => json ? JSON.parse(json) : {}
 
 // create the fetch() body
-var body = (method, data) => ({ method, body: json(data) })
+var body = (method, data) => ({method, headers: {'Content-Type':'application/json',}, body:json(data)})
```
Comment (deathbask3t) [inline fe/src/services/http.js]: make string this a constant

File: fe/src/services/ticker.js [hunk 20]
```diff
@@ -8,6 +8,7 @@
 const tickerService = {
   async getTickers() {
     // return a list of valid ticker names
+
```
Comment (deathbask3t) [inline fe/src/services/ticker.js]: spaces

File: fe/src/components/Home.vue [hunk 21]
```diff
@@ -0,0 +1,21 @@
+<template>
+  <div id="Home">
+    <h1>Home Page</h1>
+    <p>Already have an account? Login <router-link to='/login'>here</router-link></p>
+    <p>Don't have an account? Register <router-link to='/register'>here</router-link></p>
+  </div>  
+</template>
+
+<script>
+
```
Comment (deathbask3t) [inline fe/src/components/Home.vue]: spaces

File: fe/src/components/Home.vue [hunk 22]
```diff
@@ -0,0 +1,21 @@
...
+
+export default {
+  name: "Home"
+}
+
+</script>
+
+
+<style>
```
Comment (deathbask3t) [inline fe/src/components/Home.vue]: you can leave out <style> tags if you don't use them

File: fe/src/components/Login.vue [hunk 23]
```diff
@@ -1,19 +1,20 @@
 <template>
   <div class="login">
     <el-row>
-      <el-col offset=9 span=6>
+      <el-col :offset="9" :span="6">
```
Comment (deathbask3t) [inline fe/src/components/Login.vue]: what is the significance of these numbers

File: fe/src/components/Login.vue (hunk 1 above)
Comment (kentblock) [reply fe/src/components/Login.vue]: Roger that

File: fe/src/components/Login.vue (hunk 3 above)
Comment (kentblock) [reply fe/src/components/Login.vue]: Just had those in there for testing purposes...removed them now.

File: fe/src/components/Login.vue (hunk 4 above)
Comment (kentblock) [reply fe/src/components/Login.vue]: Added some notifications to both pages if you take a look.

File: fe/src/services/http.js (hunk 19 above)
Comment (kentblock) [reply fe/src/services/http.js]: 10-4

File: fe/src/services/ticker.js (hunk 20 above)
Comment (kentblock) [reply fe/src/services/ticker.js]: Heard you loud and clear on the spaces man, should all be fixed for you now

File: fe/src/components/Login.vue (hunk 23 above)
Comment (kentblock) [reply fe/src/components/Login.vue]: Those numbers specify column width and offset width, that's how it's done in the element docs, let me know if you recommend doing the layout another way.

File: fe/src/components/Register.vue (hunk 15 above)
Comment (kentblock) [reply fe/src/components/Register.vue]: Added some nav guards in main.js.

File: fe/src/main.js (hunk 18 above)
Comment (kentblock) [reply fe/src/main.js]: sorry man I will try to keep it cleaner :)

File: fe/src/components/Register.vue (hunk 13 above)
Comment (kentblock) [reply fe/src/components/Register.vue]: again, just had it in there for testing, should be a notification now

File: fe/src/components/Login.vue (hunk 9 above)
Comment (kentblock) [reply fe/src/components/Login.vue]: done

File: fe/src/components/Login.vue (hunk 8 above)
Comment (kentblock) [reply fe/src/components/Login.vue]: this.$refs['Loginform'] ?? what's the benefit of that if you don't mind me asking

**Pull request merged**